src/compile-route.js

```diff
@@ -62,6 +63,55 @@ function areHeadersEqual (currentHeader, expectedHeader) {
     return true;
 }
 
+const pathPattern = /^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/;
+const queryPattern = /^(\?)?([0-9A-Za-z]+=[0-9A-Za-z]+(\&)?)+$/;
+
+function getUrlSections (url) {
+    const splitUrl = url.split('?');
+    const pathPart = splitUrl[0];
+    const queryPart = splitUrl.length > 1 ? splitUrl.slice(1).join('?') : '';
+
+    return {
+        path: pathPart,
+        query: queryPart
+    }
+}
+
+function isValidUrl (checkedUrl) {
```

Comment:
Similar to above, if you use the URL class, there will probably be no need for this custom validation